src/openqaoa-core/tests/test_bpsp.py

```diff
@@ -53,69 +45,77 @@ def test_bpsp_terms_weights_constant(self):
         This method checks if setting such sequences raises a ValueError.
         """
+        bpsp = BPSP(np.array([0, 1, 0, 1]))
         with self.assertRaises(ValueError):
-            self.bp.car_sequence = np.array([0, 1, 0])
+            bpsp.car_sequence = np.array([0, 1, 0])
         with self.assertRaises(ValueError):
-            self.bp.car_sequence = np.array([0, 0, 1, 2, 2, 3, 3, 3])
+            bpsp.car_sequence = np.array([0, 0, 1, 2, 2, 3, 3, 3])
 
-    def test_random_instance(self):
+    def test_bpsp_random_instance(self):
         """
         Test the generation of a random BPSP instance.
 
         Validates that the generated random BPSP instance has one of the
         expected car sequences.
         """
-        instance = BPSP.random_instance(num_cars=2)
-        self.assertTrue(all(np.array([0, 1, 0, 1]) == instance.car_sequence) or 
-                        all(np.array([1, 1, 0, 0]) == instance.car_sequence) or 
-                        all(np.array([0, 0, 1, 1]) == instance.car_sequence) or 
-                        all(np.array([1, 0, 1, 0]) == instance.car_sequence))
+        bpsp = BPSP.random_instance(num_cars=2, seed=1234)
+        self.assertTrue(all(np.array([0, 1, 0, 1]) == bpsp.car_sequence) or 
+                        all(np.array([1, 1, 0, 0]) == bpsp.car_sequence) or 
+                        all(np.array([0, 0, 1, 1]) == bpsp.car_sequence) or 
+                        all(np.array([1, 0, 1, 0]) == bpsp.car_sequence))
 
-    def test_car_pos(self):
+    def test_bpsp_car_pos(self):
         """Test the retrieval of car positions."""
-        self.assertEqual(self.bp.car_positions, {0: (0, 2), 1: (1, 3)})
+        bpsp = BPSP(np.array([0, 1, 0, 1]))
+        self.assertEqual(bpsp.car_positions, {0: (0, 2), 1: (1, 3)})
 
-    def test_graph(self):
+    def test_bpsp_graph(self):
         """Test the generation of a graph representation of the BPSP instance."""
-        self.assertEqual(len(self.bp.bpsp_graph.nodes), 2)
+        bpsp = BPSP.random_instance(num_cars=2, seed=1234)
+        self.assertEqual(len(bpsp.bpsp_graph.nodes), 2)
 
-    def test_docplex_bpsp_model(self):
+    def test_bpsp_docplex_bpsp_model(self):
         """Test if the docplex model representation of BPSP is generated."""
-        self.assertIsNotNone(self.bp.docplex_bpsp_model)
+        bpsp = BPSP.random_instance(num_cars=2, seed=1234)
+        self.assertIsNotNone(bpsp.docplex_bpsp_model)
 
-    def test_cplex_solution(self):
+    def test_bpsp_cplex_solution(self):
         """
         Test the solution of the BPSP problem using CPLEX.
 
         This test assumes that CPLEX is installed and functional. It checks
         the length of the solution and the objective value.
         """
-        solution, objective_value = self.bp.cplex_solution()
+        bpsp = BPSP(np.array([0, 1, 0, 1]))
+        solution, objective_value = bpsp.solve_cplex()
         self.assertEqual(len(solution), 4)
```

Comment:
If deterministic please check the exact solution of the docplex model, if not you can ignore this

Reply:
Changed the function to check for the exact solution returned by the Docplex model.